pam/internal/pam_test/pam-client-dummy_test.go

```diff
@@ -0,0 +1,1187 @@
+package pam_test
+
+import (
```

Comment:
I don't see the point of testing a mock. Can you explain why you did this?

Reply:
Well, it's not purely a mock, as it may behave differently depending on the data passed.

And while I was hating myself for writing all the tests when doing it, I actually found various issues while doing them, so since it's used in some complex integration checks, I'd prefer to also have specific unit tests so that we're sure that it's working as expected when used for integrating in other components that they just happen to use it, but quite indirectly.